### PR TITLE
Beining/add command

### LINKED ISSUE #44: As a user, I can add new tasks to my list

### LINKED ISSUE #122: Modify Add command

### PR DESCRIPTION
**Pull request by dearvae**
fix #122 
fix #44

### CODE REVIEW

File: src/main/java/seedu/address/model/task/State.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package seedu.address.model.task;
+
+public enum State {
+    DONE("done"), PENDING("pending");
```
Comment (MarcTzh) [inline src/main/java/seedu/address/model/task/State.java:4]: Should probably be on separate lines

File: src/main/java/seedu/address/model/task/State.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    private String string;
```
Comment (MarcTzh) [inline src/main/java/seedu/address/model/task/State.java]: Should be named something more meaningful like status or taskStatus

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 3]
```diff
@@ -90,14 +90,6 @@ public void parse_compulsoryFieldMissing_failure() {
         assertParseFailure(parser, VALID_TITLE_BOB + PHONE_DESC_BOB + DESCRIPTION_DESC_BOB + ADDRESS_DESC_BOB,
                 expectedMessage);
 
-        // missing phone prefix
```
Comment (MarcTzh) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:93]: Shall discuss whether these fields should be optional in the team meeting later tonight

File: src/main/java/seedu/address/model/task/Description.java [hunk 4]
```diff
@@ -34,13 +34,27 @@ public Description(String description) {
         value = description;
     }
 
+    /**
+     * Constructs a {@code Description}.
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/model/task/Description.java]: Should have clearer documentation here

File: src/main/java/seedu/address/model/task/Phone.java [hunk 5]
```diff
@@ -26,13 +26,27 @@ public Phone(String phone) {
         value = phone;
     }
 
+    /**
+     * Constructs a {@code Phone}.
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/model/task/Phone.java]: ^

File: src/main/java/seedu/address/model/task/Status.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+ */
+public class Status {
+
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "the value is from the enum class State";
+    public final State value;
+
+    /**
+     * Constructs a {@code Status}.
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/model/task/Status.java:17]: Clearer documentation pls.

**Review by BobbyZhouZijian — CHANGES_REQUESTED** [review]
Need to fix the documentation first.

File: src/main/java/seedu/address/model/task/Description.java [hunk 7]
```diff
@@ -20,7 +20,7 @@
      * otherwise " " (a blank string) becomes a valid input.
      */
     public static final String VALIDATION_REGEX = "[\\p{Alnum},.!?][\\p{Alnum},.!? ]*";
-
+    private static final String DEFAULT_DESCRIPTION = "";
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/model/task/Description.java:23]: Why is this private variable in between to public variables?

File: src/main/java/seedu/address/model/task/Description.java [hunk 8]
```diff
@@ -34,13 +34,27 @@ public Description(String description) {
...
+
     /**
      * Returns if a given string is a valid description.
      */
     public static boolean isValidDescription(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Returns if a default Description with empty value;
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/model/task/Description.java]: I think a better documentation is : construct an empty Description when user didn't provide the description field. Caveat: Only called when the user didn't key in this field.

File: src/main/java/seedu/address/model/task/Phone.java [hunk 9]
```diff
@@ -13,6 +13,7 @@
     public static final String MESSAGE_CONSTRAINTS =
             "Phone numbers should only contain numbers, and it should be at least 3 digits long";
     public static final String VALIDATION_REGEX = "\\d{3,}";
+    private static final String DEFAULT_PHONE = "";
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/model/task/Phone.java:15]: Same. private variable in between public variables.

File: src/main/java/seedu/address/model/task/Phone.java [hunk 10]
```diff
@@ -26,13 +27,27 @@ public Phone(String phone) {
...
+
     /**
      * Returns true if a given string is a valid phone number.
      */
     public static boolean isValidPhone(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Returns a default phone object which have empty value
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/model/task/Phone.java]: Same. I think the documentation can be better.

File: src/test/java/seedu/address/testutil/TaskBuilder.java [hunk 11]
```diff
@@ -81,6 +81,14 @@ public TaskBuilder withPhone(String phone) {
         return this;
     }
 
+    /**
+     * Sets the {@code Phone} of the {@code Task} that we are building to be default phone.
```
Comment (BobbyZhouZijian) [inline src/test/java/seedu/address/testutil/TaskBuilder.java:85]: I would recommend making this documentation clearer as well. Add another line to specify that this is only used to construct an empty Phone when the field is not in the task.

File: src/test/java/seedu/address/testutil/TaskBuilder.java [hunk 12]
```diff
@@ -89,6 +97,14 @@ public TaskBuilder withDescription(String description) {
         return this;
     }
 
+    /**
+     * Sets the {@code Description} of the {@code Task} that we are building to be default description.
+     */
+    public TaskBuilder withDefaultDescription() {
```
Comment (BobbyZhouZijian) [inline src/test/java/seedu/address/testutil/TaskBuilder.java:105]: ^

**Review by BobbyZhouZijian — COMMENTED** [review]
Generally LGTM. Just some nits to fix

**Review by BobbyZhouZijian — APPROVED** [review]
LGTM now.

**Pull request merged**